File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/di/threads/JobThread.java
package com.woowrale.jroomapp.di.threads;

import com.woowrale.jroomapp.usecases.threads.JobScheduler;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

/**
 * JobScheduler implementation based on a {@link Scheduler}
 * which will execute tasks in "an unbounded thread pool".
 */
@Singleton
public class JobThread implements JobScheduler {

    @Inject
    JobThread() {}

    @Override
    public Scheduler getScheduler() {
        return Schedulers.io();
    }

}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/di/threads/JobThread.kt
package com.woowrale.kroomapp.di.threads

import com.woowrale.kroomapp.usecases.threads.JobScheduler
import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject
import javax.inject.Singleton

class JobThread @Inject internal constructor() : JobScheduler {
    override val scheduler: Scheduler
        get() = Schedulers.io()
}